lib/streamlit/elements/lib/utils.py

```diff
@@ -158,7 +166,9 @@ def _compute_element_id(
 
 def compute_and_register_element_id(
     element_type: str,
-    user_key: str | None = None,
+    *,
+    user_key: str | None,
+    form_id: str | None,
     **kwargs: SAFE_VALUES | Iterable[SAFE_VALUES],
 ) -> str:
```

Comment:
I feel we should have a test in `utils.test.py` to add some explicit tests for this function and ensure that it keeps doing what it does (e.g. putting `active_script_hash` in the `**kwargs` etc.). Dedicated unit tests help us to narrow down issues faster and make sure that we don't regress on anything by modifying the code.

Reply:
I added another generic test to ensure the usage of `active_script_hash` and `form_id`